backends/platform/ios7/ios7_osys_main.cpp

```diff
@@ -158,9 +158,11 @@ bool OSystem_iOS7::hasFeature(Feature f) {
 	switch (f) {
 	case kFeatureCursorPalette:
 	case kFeatureFilteringMode:
+#if TARGET_OS_IOS
 	case kFeatureVirtualKeyboard:
 	case kFeatureClipboardSupport:
 	case kFeatureOpenUrl:
+#endif
```

Comment:
Shouldn't this `#if TARGET_OS_IOS` only be used for `kFeatureClipboardSupport`?
From what I can see the openURL and onscreen keyboard should still work on tvOS?

And otherwise you might need to also exclude the `kFeatureVirtualKeyboard` case on tvOS in `OSystem_iOS7::setFeatureState` below.

Reply:
Yes, my mistake. I've included kFeatureOpenUrl again.